core/src/main/java/org/infinispan/interceptors/distribution/DistributionBulkInterceptor.java

```diff
@@ -70,7 +70,7 @@ public Object visitEntrySetCommand(InvocationContext ctx, EntrySetCommand comman
       protected final FlagAffectedCommand command;
 
       private BackingEntrySet(Cache cache, CacheSet<CacheEntry<K, V>> entrySet, FlagAffectedCommand command) {
-         super(cache);
+         super(cache.getAdvancedCache().withStorageMediaType());
```

Comment:
```.withStorageMediaType()``` change the request media type, not the storage media type.

Reply:
Sorry, I bungled my question. 

I meant why wouldn't the cache injected in components already have the request media type == the storage media type, so that it doesn't do any conversion? The media type of the `EncoderCache` wrapper is going to be passed as a parameter anyway.

Reply:
By default, all caches have a request type of ```application/x-java-object```. This was added in the previous issue https://issues.redhat.com/browse/ISPN-11446
Some components (query, interceptors, etc) need to operate on the data as stored, that's why this decoration was creted